src/HttpLibWebServerAdapter/Server.cpp

```diff
@@ -56,6 +56,12 @@ namespace systelab { namespace web_server { namespace httplib {
 
 		std::thread t(std::bind(&Server::runThread, this));
 		m_thread.swap(t);
+
+		std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
this is a risky check as server may evantually need more than 100 ms to start. A problem starting should be detected in the thread that tried to start and properly handle it there. When working async you need to implement sync mechanisms for these cases instead of doings sleeps.

Reply:
You are right, after talking about it offline, I decided to change it as suggested in order to make sure that the server is being started properly.